CONTRIBUTING.md

```diff
@@ -1,81 +1,120 @@
+wasm-pack test --firefox --headless -- --features wasm_test
 ```
 
-##### Macro Tests
-When adding or updating tests, please make sure you have updated the appropriate `stderr` file, which you can find [here](https://github.com/yewstack/yew/tree/master/yew-macro/tests/macro) for the `html!` macro. These files ensure that macro compilation errors are correct and easy to understand.
+You can replace `--firefox` with `--chrome` or `--safari` if you want to run the tests in a different browser. Currently Yew's CI tests use Firefox.
 
-To update or generate a new `stderr` file you can run `TRYBUILD=overwrite cargo test --test macro_test` or `TRYBUILD=overwrite cargo test --test derive_props_test` from the `yew-macro` directory.
+Make sure to run the command in the directory of the crate you wish to test.
+If you run the command in the repository root you will get an error like this:
+
+```bash
+Error: failed to parse manifest: yew/Cargo.toml
+Caused by: missing field `package`
+```
+
+#### Manually installing a webdriver
+
+`wasm-pack` automatically downloads the selected webdriver for you.
+If desired, you can download it manually. For instructions, please consult the documentation of the webdriver you wish to install.
+
+You might want to pass the argument `--mode no-install` to `wasm-pack` to make sure it doesn't try to install something.
+If the webdriver binary is in the path you can use the same command as above to run the tests.
+If not, you need to tell `wasm-pack test` where to find it using the `--geckodriver`, `--chromedriver`, or `--safaridriver` option.
+Run `wasm-pack test --help` to learn more.
+
+#### Running all tests
+
+You can use the `/ci/run_tests.sh` script to run all tests. This is the same script that is used by Yew's CI to run the whole test suite.
+The script currently always runs the tests in Firefox.
+
+#### Fetch service tests
+
+This only applies to the `yew` (and `yew-stdweb`) crate.
 
-##### Running Tests
+The tests for the fetch service require a local httpbin server. We recommend running this with [Docker](https://www.docker.com/):
 
 ```bash
-./ci/run_tests.sh
+docker run -p 8000:80 kennethreitz/httpbin
 ```
-or
+
+Before running the tests you need to set the `HTTPBIN_URL` environment variable to the url of your httpbin instance.
+If you used the previous command to start the server the value should be "http://localhost:8000" (It's important that you don't add a trailing slash).
+
+```shell
+# Unix-like
+export HTTPBIN_URL="http://localhost:8000"
+
+# Windows
+set HTTPBIN_URL=http://localhost:8000
+```
+
+You also need to activate the `httpbin_test` feature in order for the tests to run:
 
 ```bash
-cargo test --target wasm32-unknown-unknown --features wasm_test
+wasm-pack test --firefox --headless -- --features wasm_test,httpbin_test
 ```
 
-#### Benchmarks
+If you're using the `/ci/run_tests.sh` script you only need to set the environment variable. The script will automatically add the feature.
+
+### Macro Tests
```

Comment:
```suggestion
### Macro tests
```

Reply:
Is there a particular reason why you don't like capitalized titles?